### PR TITLE
Isolated label fix

### PR DESCRIPTION
**Pull request by mumichae**
Recomputed isolated label due to ineffective cluster optimisation.

The isolated label score now optimises cluster resolution by F1 score between cluster assignment and isolated cell label. This is done for each isolated cell label separately.

### CODE REVIEW

File: scIB/metrics.py [hunk 1]
```diff
@@ -361,35 +360,44 @@ def get_isolated_labels(adata, label_key, batch_key, cluster_key, n, verbose):
...
-    import sklearn.metrics as scm
     adata_tmp = adata.copy()
-    
-    # cluster optimizing over cluster with largest number of isolated label per batch
+
     def max_label_per_batch(adata, label_key, cluster_key, label, argmax=False):
+        """cluster optimizing over cluster with largest number of isolated label per batch"""
         sub = adata.obs[adata.obs[label_key] == label].copy()
         label_counts = sub[cluster_key].value_counts()
+        print(label_counts)
```
Comment (LuckyMD) [inline scIB/metrics.py:376]: this looks like it shouldn't be turned on by default but only if verbose or sth, no?

File: scIB/metrics.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import pandas as pd
 import matplotlib.pyplot as plt
 import seaborn as sns
+import sklearn
```
Comment (LuckyMD) [inline scIB/metrics.py:7]: do you need this? Or does sklearn.metrics suffice?

**Review by LuckyMD — APPROVED** [review]
looks good overall.. just 2 small things.

File: scIB/metrics.py (hunk 1 above)
Comment (mumichae) [reply scIB/metrics.py:376]: Yep, removed those

File: scIB/metrics.py (hunk 2 above)
Comment (mumichae) [reply scIB/metrics.py]: We also want to use `sklearn.linear_model` for the PCR score, so I'd keep it the way it is. Also, it's nicer to see `sklearn.metrics` rather than `metrics` so we know exactly where that function comes from. I removed all imports from within functions, as they would only cause hidden errors

Comment (LuckyMD) [reply scIB/metrics.py]: okay

Comment (danielStrobl) [reply scIB/metrics.py]: this doesn't work. sklearn doesn't automatically import subpackages. I think you need an `import sklearn.metrics` there as well
